### PR TITLE
feat: render opcodes diff

### PR DESCRIPTION
**Pull request by leovct**
Related to #6 

This PR merges the work done in #13  and #14. It was simpler to create a new PR than to rebase them.

- Introduce a new component `DiffOpcodes` that renders the diff between chain opcodes. Screenshots of the rendering are available in the description of #14.
- Update the logic of `pages/diff.tsx` to render the proper `SectionComponent` using `SECTION_MAP`. It also enables to hide some sections such as the `DiffOpcodes` for now.
- Create a new `types` module and create a file for each important "type" (e.g. `Chain`, `Opcode`, etc.).
- Add a new `Reference` type in `Opcode` type to add a bit of context to a url via a `title`.
- Comments from #14 have been taken into account.

### CODE REVIEW

File: src/types/opcode.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  description: string;
+  minGas: number;
+  gasComputation?: GasComputation;
+  inputs?: Variable[];
+  outputs?: Variable[];
+  examples: Example[];
+  playgroundLink?: string;
+  errorCases: string[];
+  notes?: string[];
+  references: Reference[];
```
Comment (mds1) [inline src/types/opcode.ts:42]: I was thinking of introducing `type Markdown = string`, and adding a simple markdown parser to the UI components. We can leave this as-is for this PR, but just wanted to share how I was thinking about things. Being able to use markdown in these arbitrary string fields (reference, notes, error cases, etc.) will give a lot more flexibility on how the UI side, while keeping a decent separation of concerns between the content itself and how it's rendered (it of course introduces some coupling but it seems like an ok tradeoff, still thinking about it though)

File: src/pages/diff.tsx [hunk 2]
```diff
@@ -1,16 +1,28 @@
...
-  signatureTypes: 'Transaction and Signature Types',
+interface Props<T> {
+  base: T;
+  target: T;
+  onlyShowDiff: boolean;
+}
+
+const SECTION_MAP: Record<
+  string,
+  { title: string; component: React.ComponentType<Props<any>>; hide?: boolean }
```
Comment (mds1) [inline src/pages/diff.tsx:20]: Nice cleanup here and with the `SectionComponent` 👌

**Review by mds1 — APPROVED** [review]
LGTM, thank you!

File: src/types/opcode.ts (hunk 1 above)
Comment (leovct) [reply src/types/opcode.ts:42]: Hey, that's something I was also thinking about! 😄 I'm not familiar with Markdown parser and stuff but this sounds like a great solution to have more flexibility.

**Pull request merged**